gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -530,6 +529,44 @@ private void pollAndAdvanceDag()
       }
     }
 
+    private ExecutionStatus getJobExcecutionStatus(boolean slaKilled, JobStatus jobStatus) {
+      if (slaKilled) {
+        return CANCELLED;
+      } else {
+        if (jobStatus == null) {
+          return PENDING;
+        } else {
+          return valueOf(jobStatus.getEventName());
+        }
+      }
+    }
+
+    /**
+     * Check if the SLA is configured for this job. If it is, tries to cancel the job if SLA is reached.
+     * @param node dag node of the job
+     * @return true if the job is killed because it reached sla
+     * @throws ExecutionException exception
+     * @throws InterruptedException exception
+     */
+    private boolean slaKillIfNeeded(DagNode<JobExecutionPlan> node) throws ExecutionException, InterruptedException {
+      long flowStartTime = DagManagerUtils.getFlowStartTime(node);
+      long currentTime = System.currentTimeMillis();
+      long flowSla = DagManagerUtils.getFlowSla(node);
+
+      if (flowSla != -1L && currentTime > flowStartTime + flowSla) {
+        log.info("Job exceeded the SLA of {} ms. Killing it now...", flowSla);
+        cancelDag(DagManagerUtils.generateDagId(node));
+        if (this.eventSubmitter.isPresent()) {
+          JobExecutionPlan jobExecutionPlan = DagManagerUtils.getJobExecutionPlan(node);
+          Map<String, String> jobMetadata = TimingEventUtils.getJobMetadata(Maps.newHashMap(), jobExecutionPlan);
+          this.eventSubmitter.get().getTimingEvent(TimingEvent.LauncherTimings.JOB_CANCEL).stop(jobMetadata);
```

Comment:
I don't think sending JOB_CANCEL timer event is right. We should be sending a new timer event FLOW_CANCEL since we have cancelled the entire flow.

Reply:
Fixed it, cancelling only one node not the whole dag